### PR TITLE
2197 - 1

### PR DESCRIPTION
**Pull request by shamritskiy3468**
## Номер
2197

## Номер задания
1

## Ссылка на видео с демо
https://www.youtube.com/watch?v=XSYP5CtL2xo

## Комментарии
RUBY - KOANS

### CODE REVIEW

File: 2197/1/koans/about_dice_project.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# Implement a DiceSet Class here:
+# :reek:Attribute
+class DiceSet
+  attr_reader :values
+  attr_writer :values
+
+  def new; end
```
Comment (allomov) [inline 2197/1/koans/about_dice_project.rb]: NO. не надо.

File: 2197/1/koans/about_dice_project.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  def roll(numb_of_rolls)
+    @values = (0..(numb_of_rolls - 1)).to_a.map { rand(1..6) }
+  end
```
Comment (allomov) [inline 2197/1/koans/about_dice_project.rb]: можно красивее :)

File: 2197/1/koans/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+# :reek:FeatureEnvy
+# :reek:UncommunicativeParameterName
+def triangle(a, b, c)
+
+	if a < 0
```
Comment (allomov) [inline 2197/1/koans/triangle.rb]: создай метод `raise_if_not_triangle_with_sides(...)`

File: 2197/1/koans/triangle.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		raise TriangleError
+	elsif a + b <= c or a + c <= b or b + c <= a
```
Comment (allomov) [inline 2197/1/koans/triangle.rb]: read this and change code for the method:
- https://github.com/rubocop-hq/ruby-style-guide

File: 2197/1/koans/triangle.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		return :equilateral 
+	elsif b_side == a_side || b_side == c_side || c_side == a_side
+		return :isosceles
+	end
+	:scalene
+end
+# :reek:FeatureEnvy
+# :reek:UncommunicativeParameterName
+def check_if_triangle_is_valid(a_side, b_side, c_side)
+	if (a_side <= 0 || b_side <= 0 || c_side <= 0) || (a_side + b_side <= c_side || a_side + c_side <= b_side || b_side + c_side <= a_side)
```
[the remark this answers is not in the document]
Comment (nplusp) [reply 2197/1/koans/triangle.rb]: And finally, ideally, you would like to split these validations and provide a message to these errors.

It's pretty easy thing to do:

raise TriangleError, 'Side length cannot be less than a zero!' if ...
raise TriangleError, 'Sum of two lesser sides cannot be less that third side!'

**Pull request merged**